pkg/aws/manager_rds_subnet_group.go

```diff
@@ -0,0 +1,105 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/integr8ly/cluster-service/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	loggingKeySubnetGroup = "subnet-group-name"
+)
+
+var _ ClusterResourceManager = &RDSSubnetGroupManager{}
+
+type RDSSubnetGroupManager struct {
+	rdsClient rdsClient
+	logger    *logrus.Entry
+}
+
+func NewDefaultRDSSubnetGroupManager(session *session.Session, logger *logrus.Entry) *RDSSubnetGroupManager {
+	fmt.Println("creating new RDS Subnet Group Manager")
+	return &RDSSubnetGroupManager{
+		rdsClient: rds.New(session),
+		logger:    logger.WithField("engine", managerRDS),
+	}
+}
+
+func (r *RDSSubnetGroupManager) GetName() string {
+	return "AWS RDS Subnet Group Manager"
+}
+
+// Delete all RDS Subnet Groups for a specified cluster
+func (r *RDSSubnetGroupManager) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	r.logger.Debug("deleting resources for cluster")
+	subnetGroupsDescribeInput := &rds.DescribeDBSubnetGroupsInput{}
+	subnetGroupsDescribeOutput, err := r.rdsClient.DescribeDBSubnetGroups(subnetGroupsDescribeInput)
```

Comment:
Any way we can reuse the resource tagging API here? It may help us avoid some network requests e.g.

- List all resources that are RDS instances with the cluster tag
- Add those to the deletion list

See [1] for an example

[1] https://github.com/integr8ly/cluster-service/blob/master/pkg/aws/manager_s3.go#L55

Reply:
Sure thing, I can try that. I mostly just copied what I found in the RDS Manager file 😅

Reply:
Pushed a new commit implementing this idea (with unit tests)

Reply:
@darahayes Yep the RDS Manager is a little out of date compared to the others. Thanks for making the change